⬆️ Upgrade Starlette supported version range to >=0.40.0,<0.49.0

musicinmybrain opened the pull request:
Starlette 0.48.0 introduced `HTTP_422_UNPROCESSABLE_CONTENT` as a new name for `HTTP_422_UNPROCESSABLE_ENTITY` and deprecated the old name due to RFC 9110; see https://github.com/Kludex/starlette/pull/2939.

I updated all instances ~to the new name and increased the lower bound on the Starlette version~ to use the literal integer or string value `422`/`"422"` instead of either `HTTP_422_UNPROCESSABLE_CONTENT` or `HTTP_422_UNPROCESSABLE_ENTITY`, which makes FastAPI compatible with Starlette >=0.48 without deprecation warnings while preserving backwards compatibility with Starlette<0.48.

@Kludex, FYI.

musicinmybrain committed:
⬆️ Upgrade Starlette supported version range to >=0.48.0,<0.49.0

musicinmybrain commented:
> If we use 429 instead of the constant, there's no need to bump.
> 
> Maybe we should do that?

As in, just write literal integer `422`? Sure, that’s a sensible idea, and it would be easy enough to do. @tiangolo, do you have a preference one way or the other?

tiangolo commented:
Yep, I think a literal `422` should be enough, and that way we avoid having to restrict the lower bound limit, which is otherwise not necessary. :nerd_face:

musicinmybrain committed:
⬆️ Expand Starlette supported version range to >=0.40.0,<0.49.0

Restore support for Starlette 0.40 through 0.47 (without deprecation
warnings in Starlette 0.48) by using the literal integer `422` or string
representation `"422"` in place of either the new RFC 9110 name
`HTTP_422_UNPROCESSABLE_CONTENT` or the earlier RFC 7231 name
`HTTP_422_UNPROCESSABLE_ENTITY`.

svlandeg committed:
remove comments

tiangolo reviewed (approved):
Nice, thank you! 🚀 

And thanks @Kludex and @svlandeg! 🙌 

This will be available in FastAPI 0.116.2, released in the next hours.
